R/parse_params.R

```diff
@@ -38,29 +38,32 @@ parse_params <- function(station_id, start, end, hour = FALSE) {
   # Parse Dates -------------------------------------------------------------
 
   if (is.null(start) & is.null(end)) {
+    #API is always about one timestep behind
```

Comment:
For `az_hourly`, the code `az_hourly("az01", start_date_time = "2023-10-25")`. gives the following feedback and error:

`Querying data through 2023-10-24 23:59:59`

`Error in parse_params(station_id = station_id, start = start_date_time,  :  end_date is before start_date!`

A similar error occurs with `az_daily("az01", start_date = "2023-10-25")`

So, when `end_date_time` and `end_date` are not specified, the default is `today - 1`. 

But, when `end_date_time` and `end_date` are specified as `today` or `tomorrow` or whenever in the future, feedback instead is, for example:

`az_hourly("az01", start_date_time = "2023-10-22", end_date_time = "2023-10-26")`

`Querying data through 2023-10-26 23:59:59
Warning messages:
1: In az_hourly("az01", start_date_time = "2023-10-22", end_date_time = "2023-10-26") :
  Some requested data were unavailable
2: In az_hourly("az01", start_date_time = "2023-10-22", end_date_time = "2023-10-26") :
  You requested data through 2023-10-26 23:59:00 but only data through 2023-10-25 10:00:00 were available`

What do you think about setting the default assignment for when `end_date_time` and `end_date` are not specified as `today`?

Reply:
For `az_daily`, this likely is moot, as someone querying daily data for today when today isn't done yet doesn't make sense. But for `az_hourly`, it would allow someone to query the latest hourly data, which is about one time step behind current time, as you comment here (above).

Reply:
ack! dang it, I thought I fixed that one.  I'll write some tests and try again.

Reply:
OK. I don't think we need to silence the message `Querying data from {start} to {end}`. It's good feedback whether or not `{start}` or `{end}` are provided. It either sets expectation or confirms it, which is good in either instance.

Reply:
Working on making parse_params a lot clearer and implementing these messages.  I've run into a situation where requesting the most recent data with `az_hourly(start=NULL, end = NULL)` prints this:

```
Querying data from 2023-10-26 07:00 to 2023-10-26 07:00
Warning message:
In az_hourly() :
  You requested data through 2023-10-26 07:00 but only data through 2023-10-25 23:59 were available
```
And the data returned is actually from 2023-10-25 01:00: to 2023-10-25 23:59
The issue is that these dates aren't actually what gets passed to the API.  The actual request looks like `https://api.azmet.arizona.edu/v1/observations/hourly/*/*/*`
My initial thoughts are to change the message to something like "Querying most recent data" when start and end aren't supplied and not show a warning.  Similarly, if only a start date is supplied, print a message "Querying data since {start_date}".   What do you think?

Reply:
I'll make a matrix of the different combinations of `start` and `end` argument possibilities with corresponding messages